src/main/java/apoc/util/JsonUtil.java

```diff
@@ -73,10 +77,13 @@ public void close() throws IOException {
     public static <T> T parse(String json, String path, Class<T> type) {
         if (json==null || json.isEmpty()) return null;
         try {
-            if (path == null || path.isEmpty()) return OBJECT_MAPPER.readValue(json, type);
-            return JsonPath.parse(json,JSON_PATH_CONFIG).read(path,type);
+            if (path == null || path.isEmpty()) {
+                return OBJECT_MAPPER.readValue(json, type);
+            }
+            String stringValue = OBJECT_MAPPER.writeValueAsString(JsonPath.parse(json,JSON_PATH_CONFIG).read(path, type));
```

Comment:
Can we please not do this, and find a way to avoid creating new huge string objects in memory?

Reply:
Done we now inject Jackson into JsonPath

Reply:
Great!